### PR TITLE
Restructure README, add CONTRIBUTING, move `example-consumers` to `consumers`

### PR DESCRIPTION
**Pull request by rji**
This PR accomplishes the following:
  * Restructure the README
  * Add a contributing file with guidelines for community members
  * Move the `example-consumers/` directory back to `consumers/`
  * Relocates documentation to its own dedicated `docs/` directory

The idea on renaming `example-consumers` to `consumers` is that we will maintain these
consumers so that this project is capable of shipping metrics to third-party systems without
additional development effort by the user/customer. That means that we'll need to add tests
and add them to our build scripts.

@dcos/day2ops please check out the documentation changes for clarity. Note that this is only
the first pass at restructuring the docs; I have additional changes that I'll make in future PRs.

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -0,0 +1,63 @@
+# Contributing
```
Comment (malnick) [inline CONTRIBUTING.md]: :+1: on contributing!!

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+### Trivial changes
+For trivial changes and documentation updates, it's not always necessary to create a new ticket
+or topic branch. Please, use your best judgement here.
+
+### Style guidelines
+Before submitting your pull request, please consider the following:
+  * Check for unnecessary whitespace by running `git diff --check`.
+  * Check for syntax and style issues by running `go fmt`.
+  * Include comments and examples where appropriate.
+  * No additional copyright statements or licenses.
```
Comment (malnick) [inline CONTRIBUTING.md]: We should add in our preferred golang linters and ensure appropriate formatting.

Comment (malnick) [inline CONTRIBUTING.md]: And maybe add a styleguide for the module code (c++) and the consumers (java)

**Comment by rji** [comment]
Thanks for the review @malnick! Addressed your comments (added style guide and linter) to the contributing docs.

**Pull request merged**